### DISCUSSION TITLE
Using K8sRunLauncher and CeleryK8sRunLauncher in Helm deployment

### DISCUSSION THREAD

**Discussion by peter-lim-yello** [dagster-io/dagster#19440] [Q&A] [answered]
Hello,

Are we able to use both K8sRunLauncher and CeleryK8sRunLauncher in a single helm deploy? We have two user code deployments, user_code_1 needing K8sRunLauncher, and user_code_2 needing CeleryK8sRunLauncher.

Thank you!

**Comment by prha** [comment, marked as answer]
Run launching is configured at the instance level, so the only way to get customizable behavior is to use a custom run launcher.

You could implement your own custom run launcher, which internally instantiates an instance of the k8s run launcher and the celery k8s run launcher.  You could then determine on a run-by-run basis which launcher to use (maybe by using tags, job_name, or some other heuristic) in the `launch_run` method.

This approach is also mentioned in this discussion: https://github.com/dagster-io/dagster/discussions/15856#discussioncomment-6732093

**Comment by peter-lim-yello** [comment]
Thank you! Is there any plan to make this officially supported by Dagster?

**Reply by prha** [reply]
I don't think there are any plans to make this more supported than it already is.  The logic of how to route runs is going to be customized on an instance-by-instance basis.
